File: r/perfect-numbers/perfect-numbers.R
number_type <- function(n){
  if (n > 1) {
    possible.factors <- c(1:(n - 1))
    aliquot.sum <- sum(possible.factors[n %% possible.factors == 0])
  } else if (n == 1) {
    aliquot.sum <- 0
  } else {
    stop("Invalid number: ", n)
  }
  if (aliquot.sum < n) {
    "deficient"
  } else if (aliquot.sum > n) {
    "abundant"
  } else {
    "perfect"
  }
}

File: javascript/perfect-numbers/perfect-numbers.js
export function classify(num) {
    if (num <= 0) {
        throw new Error('Classification is only possible for natural numbers.');
    }
    let aliquot = 0;
    for (let i = 1; i < num; i++) {
        if (num % i == 0) {
            aliquot += i;
        }
    }
    if (aliquot < num) {
        return 'deficient';
    } else if (aliquot > num) {
        return 'abundant';
    } else {
        return 'perfect';
    }
}
